pages/Identity/Registry.md

```diff
@@ -6,21 +6,228 @@ menu: Identity
 
 # Identity Registry
 
+A registry allows for distinct user identifiers and human readable discovery of network members.
+The DSNP Id Registry is a simple contract that allows switching handles, identity contracts, and chain migration
+while maintaining all graph connections, public and private.
+
 ## Specification Status
 
 | Version | Status |
 ---------- | ---------
-| 0.1     | Draft |
+| 0.1     | Tentative |
 
 ## Purpose
-1. Describe how an Identity Registry resolves a name or number to an identity contract
+1. Describes how the Identity Registry resolves a DSNP Id to an identity contract address
+1. Describes how the Identity Registry allows for handle resolution
+1. Presents the interface for the Identity Registry
+1. Describes rejected alternatives
 
 ## Assumptions
 
-TBD
+* Ids will need to be moved from Betanet to Mainnet
+* Handles are for display and discovery purposes only
+
+## Discovery via DSNP Handles
+
+Contract addresses or numerical ids are not easy to remember.
+Most networks rely on text based handles for discovery of users on a network.
+DSNP handles are an easy way to allow easy user lookup.
+
+### Handles
+
+Handles are simple UTF-8 strings.
+No limitations are placed on length or contents, although different clients may not have support for the full unicode set.
+A user MAY NOT register multiple handles that point to the same DSNP Id.
+Handles must be unique.
+
+### Homograph Attack Mitigation
+
+UTF-8 support for handles opens handle users up to [homograph attacks](https://en.wikipedia.org/wiki/IDN_homograph_attack), not to mention case-sensitivity issues.
+This issue is of ongoing discussion both for the DSNP as well as in for ICANN domain names and other projects working with internationalization support.
+
+Because the DSNP Id is stable, attacks would only be successful in cases where the DSNP Id were unknown.
+[Punycode](https://en.wikipedia.org/wiki/Punycode) would resolve some issues, but would require losing true internationalization.
```

Comment:
Cleaned up:

> [Punycode](https://en.wikipedia.org/wiki/Punycode) is used by some software to prevent homographs by encoding all non-Latin characters into Latin characters.
So punycode does not properly present non-Latin characters which isn't reaching the level of internationalization support desired by the DSNP.